src/BaGet.Core/Services/PackagesReconciler.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace BaGet.Core.Services
+{
+    public class PackagesReconciler
+    {
+        private readonly IStorageService _storage;
+        private readonly IPackageIndexingService _index;
+        private readonly ILogger<PackageIndexingService> _logger;
+
+        public PackagesReconciler(
+            IStorageService storage,
+            IPackageIndexingService index,
+            ILogger<PackageIndexingService> logger)
+        {
+            _storage = storage;
+            _index = index;
+            _logger = logger;
+        }
+
+        public async Task Reconcile(CancellationToken cancellationToken = default)
+        {
+            var allStored = await _storage.GetPackagePathsAsync(cancellationToken);
```

Comment:
Let's not use `IStorageService` here, I'm okay with only supporting the local filesystem.